### ISSUE TITLE
[Feature Request] Env checker for VecEnv

### ISSUE THREAD

**Issue by AlexPasqua** [DLR-RM/stable-baselines3#1251]
### 🚀 Feature

Check the environment when creating a `VecEnv`

### Motivation

I noticed that [`check_env`](https://github.com/DLR-RM/stable-baselines3/blob/2bb8ef5e632a0e0dda291c2cd6735da75a4fcb7e/stable_baselines3/common/env_checker.py#L319) doesn't work with `VecEnv`'s (#653), but I think it would be a good idea to allow to check the correctness of vectorized envs too.

One could do it manually, after the creation of the base env, but if a user calls [`make_vec_env`](https://github.com/DLR-RM/stable-baselines3/blob/2bb8ef5e632a0e0dda291c2cd6735da75a4fcb7e/stable_baselines3/common/env_util.py#L38) to create a vectorized env and passes a function as `env_id` (because the env might have multiple wrappers), the base env is created automatically within that function and the user cannot check the correctness of its implementation.
For example, on the model of #894 ([this comment](https://github.com/DLR-RM/stable-baselines3/issues/894#issuecomment-1261329620)), I do the following:
```python
def make_env():
    env = MyEnv()
    env = TimeLimit(env, 100)  # new time limit
    env = MyWrapper(env)
    env = MyOtherWrapper(env)
    return env

vec_env = make_vec_env(env_id=make_env, ...)
```
So the user has to remember every time to check its env within `make_env`.

### Pitch

Check env inside [`make_vec_env`](https://github.com/DLR-RM/stable-baselines3/blob/2bb8ef5e632a0e0dda291c2cd6735da75a4fcb7e/stable_baselines3/common/env_util.py#L38)

#### Alternative 1 (my favourite)
Since [`make_vec_env`](https://github.com/DLR-RM/stable-baselines3/blob/2bb8ef5e632a0e0dda291c2cd6735da75a4fcb7e/stable_baselines3/common/env_util.py#L38) is a library call, we can assume it works correctly. Therefore I would check the env automatically **before** vectorizing it, by adding a call to [`check_env`](https://github.com/DLR-RM/stable-baselines3/blob/2bb8ef5e632a0e0dda291c2cd6735da75a4fcb7e/stable_baselines3/common/env_checker.py#L319) within [`make_vec_env`](https://github.com/DLR-RM/stable-baselines3/blob/2bb8ef5e632a0e0dda291c2cd6735da75a4fcb7e/stable_baselines3/common/env_util.py#L38).

#### Alternative 2
Make [`check_env`](https://github.com/DLR-RM/stable-baselines3/blob/2bb8ef5e632a0e0dda291c2cd6735da75a4fcb7e/stable_baselines3/common/env_checker.py#L319) work also with vectorized environments.

### Additional context

I could implement it if you agree.

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo

**Comment by araffin** [comment]
Hello,

> but I think it would be a good idea to allow to check the correctness of vectorized envs too.

yes, would be a good idea =)

> but if a user calls [make_vec_env](https://github.com/DLR-RM/stable-baselines3/blob/2bb8ef5e632a0e0dda291c2cd6735da75a4fcb7e/stable_baselines3/common/env_util.py#L38) to create a vectorized env and passes a function as env_id (because the env might have multiple wrappers), the base env is created automatically within that function and the user cannot check the correctness of its implementation.

I think we are talking about different things.
If a single env work with `env_checker`, then it will work with `make_vec_env`.
I had more checking vectorized env by design in mind (envpool, Isaac gym, or https://github.com/DLR-RM/stable-baselines3/issues/1235) where you cannot access a single env separately.

> Check env inside [make_vec_env](https://github.com/DLR-RM/stable-baselines3/blob/2bb8ef5e632a0e0dda291c2cd6735da75a4fcb7e/stable_baselines3/common/env_util.py#L38)

checking by default is a bad idea in my opinion as you will unexpectedly call `reset` and `step` for that (unexpected behavior/calls to reset/step is not nice for the user).

**Comment by AlexPasqua** [comment]
> If a single env work with `env_checker`, then it will work with `make_vec_env`.

Yes, but the user has to remember to check it before vectorizing it. I though it would be handy to have a function that takes an env (or a function returning an env), checks it, and creates a vectorized environment, all automatically. Since SB3 already has `make_vec_env`, I though to put the ckecking in it.


> checking by default is a bad idea in my opinion as you will unexpectedly call `reset` and `step` for that (unexpected behavior/calls to reset/step is not nice for the user).

It would be possible to pass a flag to [`make_vec_env`](https://github.com/DLR-RM/stable-baselines3/blob/2bb8ef5e632a0e0dda291c2cd6735da75a4fcb7e/stable_baselines3/common/env_util.py#L38) so that the user can decide whether to check the env ("at the cost of" calling `reset` and `step` ) or not.


> I had more checking vectorized env by design in mind (envpool, Isaac gym, or #1235) where you cannot access a single env separately.

Yes, we were probably talking about different things then 😅
